File: Leetcode/701-Insert-Into-A-Binary-Search-Tree/InsertIntoABinarySearchTree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func insertIntoBST(root *TreeNode, val int) *TreeNode {
	node := &TreeNode{Val: val}
	if root == nil {
		root = node
		return root
	}
	cur := root
	for true {
		if cur.Val > val {
			if cur.Left == nil {
				cur.Left = node
				break
			}
			cur = cur.Left
			continue
		}
		if cur.Right == nil {
			cur.Right = node
			break
		}
		cur = cur.Right
	}
	return root
}

File: Leetcode/701-Insert-Into-A-Binary-Search-Tree/InsertIntoABinarySearchTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function(root, val) {
    var node = new TreeNode(val)
    if (!root) {
        root = node
        return root
    }
    var cur = root
    while (true) {
        if (cur.val >val) {
            if (!cur.left) {
                cur.left = node
                break
            }
            cur = cur.left
            continue
        }
        if (!cur.right) {
            cur.right = node
            break
        }
        cur = cur.right
    }
    return root
};
